### PR TITLE
Updated the documentation of `returnDecisionReason`

### PR DESCRIPTION
**Pull request by rinormaloku**
Updated the documentation of `returnDecisionReason`

### CODE REVIEW

File: projects/gloo/api/v1/enterprise/options/extauth/v1/extauth.proto [hunk 1]
```diff
@@ -1199,16 +1199,20 @@ message OpaAuthOptions {
...
-    // When using OpaServerAuth, the entire document will be returned as metadata.
-    // The `allowed` field on the document is used to make the policy decision.
-    // The entry will be in the returned DynamicMetadata in the CheckResponse with the structure
-    // `envoy.filters.http.ext_authz:
-    //   -> name of the auth step`, such as `spec.configs[i].name
-    //       -> reason`.
-    // If set to false, the response is allowed or denied based on the Rego rules without returning the reason.
+    // DEPRECATED: It's recommended to use the `dynamic_metadata` field within Rego policies to specify the decision reason. To learn more about this approach, see the [OPA Envoy Plugin docs](https://github.com/open-policy-agent/opa/blob/c12463c/docs/content/envoy-primer.md#example-policy-with-additional-controls).
+    //
+    // When `returnDecisionReason` is set to true the decision reason is stored in the Envoy Dynamic Metadata and has the following properties:<ul>
```
Comment (Nadine2016) [inline projects/gloo/api/v1/enterprise/options/extauth/v1/extauth.proto:1204]:
```suggestion
    // When `returnDecisionReason` is set to true, the decision reason is stored in the Envoy Dynamic Metadata and has the following properties:<ul>
```

File: projects/gloo/api/v1/enterprise/options/extauth/v1/extauth.proto [hunk 2]
```diff
@@ -1199,16 +1199,20 @@ message OpaAuthOptions {
...
+    // <li>`body` - a textual explanation of the decision</li>
+    // <li>`allowed` - whether the request was allowed or rejected</li></ul>
+    //
+    // When using OpaAuth, the `body` field must be the second parameter of the query.
+    //
+    // Both approaches OpaAuth and OpaServerAuth use the `allowed` and `body` values from the OPA response stored in the Envoy Dynamic Metadata as described earlier.
+    // In the CheckResponse these values are nested under `envoy.filters.http.ext_authz`, followed by the authentication step's name (e.g., `spec.configs[i].name`),
+    // and then under `reason`, where you'll find the two fields `body` and `allowed`.
+    //
+    // If, however, returnDecisionReason is set to false, OPA's decision to allow or reject a request will be made according to the Rego policy rules, and no explanation will be provided.
```
Comment (Nadine2016) [inline projects/gloo/api/v1/enterprise/options/extauth/v1/extauth.proto:1214]:
```suggestion
    // If, however, returnDecisionReason is set to false, OPA's decision to allow or reject a request is made according to the Rego policy rules, and no explanation is provided.
```

File: projects/gloo/api/v1/enterprise/options/extauth/v1/extauth.proto [hunk 3]
```diff
@@ -1199,16 +1199,20 @@ message OpaAuthOptions {
...
+    // Despite this, the `dynamic_metadata` field can still be used to convey any necessary information to the Envoy Dynamic Metadata, including the decision reason.
```
Comment (Nadine2016) [inline projects/gloo/api/v1/enterprise/options/extauth/v1/extauth.proto:1215]:
```suggestion
    // Despite of this, the `dynamic_metadata` field can still be used to convey any necessary information to the Envoy Dynamic Metadata, including the decision reason.
```

File: projects/gloo/api/v1/enterprise/options/extauth/v1/extauth.proto [hunk 4]
```diff
@@ -1199,16 +1199,20 @@ message OpaAuthOptions {
...
-    // If set to false, the response is allowed or denied based on the Rego rules without returning the reason.
+    // DEPRECATED: It's recommended to use the `dynamic_metadata` field within Rego policies to specify the decision reason. To learn more about this approach, see the [OPA Envoy Plugin docs](https://github.com/open-policy-agent/opa/blob/c12463c/docs/content/envoy-primer.md#example-policy-with-additional-controls).
+    //
+    // When `returnDecisionReason` is set to true the decision reason is stored in the Envoy Dynamic Metadata and has the following properties:<ul>
+    // <li>`body` - a textual explanation of the decision</li>
+    // <li>`allowed` - whether the request was allowed or rejected</li></ul>
+    //
+    // When using OpaAuth, the `body` field must be the second parameter of the query.
+    //
+    // Both approaches OpaAuth and OpaServerAuth use the `allowed` and `body` values from the OPA response stored in the Envoy Dynamic Metadata as described earlier.
```
Comment (Nadine2016) [inline projects/gloo/api/v1/enterprise/options/extauth/v1/extauth.proto:1210]:
```suggestion
    // Both the OpaAuth and OpaServerAuth approaches use the `allowed` and `body` values from the OPA response that are stored in the Envoy Dynamic Metadata as described earlier.
```

File: projects/gloo/api/v1/enterprise/options/extauth/v1/extauth.proto [hunk 5]
```diff
@@ -1199,16 +1199,20 @@ message OpaAuthOptions {
...
+    // In the CheckResponse these values are nested under `envoy.filters.http.ext_authz`, followed by the authentication step's name (e.g., `spec.configs[i].name`),
```
Comment (Nadine2016) [inline projects/gloo/api/v1/enterprise/options/extauth/v1/extauth.proto:1211]:
```suggestion
    // In the CheckResponse, these values are nested under `envoy.filters.http.ext_authz`, and are followed by the name of the authentication step, such as  `spec.configs[i].name`. Review 
```

File: projects/gloo/api/v1/enterprise/options/extauth/v1/extauth.proto [hunk 6]
```diff
@@ -1199,16 +1199,20 @@ message OpaAuthOptions {
...
+    // and then under `reason`, where you'll find the two fields `body` and `allowed`.
```
Comment (Nadine2016) [inline projects/gloo/api/v1/enterprise/options/extauth/v1/extauth.proto:1212]:
```suggestion
    // the `reason` section to find the two fields `body` and `allowed`.
```

File: projects/gloo/api/v1/enterprise/options/extauth/v1/extauth.proto [hunk 7]
```diff
@@ -1199,16 +1199,19 @@ message OpaAuthOptions {
...
+    // DEPRECATED: It's recommended to use the `dynamic_metadata` field within Rego policies to specify the decision reason. To learn more about this approach, see the [OPA Envoy Plugin docs](https://github.com/open-policy-agent/opa/blob/c12463c/docs/content/envoy-primer.md#example-policy-with-additional-controls).
+    //
+    // When `returnDecisionReason` is set to true, the decision reason is stored in the Envoy Dynamic Metadata and has the following properties:<ul>
+    // <li>`body` - a textual explanation of the decision</li>
+    // <li>`allowed` - whether the request was allowed or rejected</li></ul>
+    //
+    // When using OpaAuth, the `body` field must be the second parameter of the query.
+    //
+    // Both the OpaAuth and OpaServerAuth approaches use the `allowed` and `body` values from the OPA response in the decision reason.
+    // You can find the `body` and `allowed` fields in the Envoy Filter Dynamic Metadata under `envoy.filters.http.ext_authz.<authentication_step_name>.reason` section.
```
Comment (Nadine2016) [inline projects/gloo/api/v1/enterprise/options/extauth/v1/extauth.proto:1211]:
```suggestion
    // You can find the `body` and `allowed` fields in the Envoy Filter Dynamic Metadata under the `envoy.filters.http.ext_authz.<authentication_step_name>.reason` section.
```

File: projects/gloo/api/v1/enterprise/options/extauth/v1/extauth.proto [hunk 8]
```diff
@@ -1199,16 +1199,19 @@ message OpaAuthOptions {
...
+    // <li>`body` - a textual explanation of the decision</li>
+    // <li>`allowed` - whether the request was allowed or rejected</li></ul>
+    //
+    // When using OpaAuth, the `body` field must be the second parameter of the query.
+    //
+    // Both the OpaAuth and OpaServerAuth approaches use the `allowed` and `body` values from the OPA response in the decision reason.
+    // You can find the `body` and `allowed` fields in the Envoy Filter Dynamic Metadata under the `envoy.filters.http.ext_authz.<authentication_step_name>.reason` section.
+    //
+    // If, however, `returnDecisionReason` is set to false, OPA's decision to allow or reject a request is made according to the Rego policy rules, and no explanation is provided.
+    // Despite of this, the `dynamic_metadata` field can still be used to convey any necessary information to the Envoy Dynamic Metadata, including the decision reason.
```
Comment (Nadine2016) [inline projects/gloo/api/v1/enterprise/options/extauth/v1/extauth.proto:1214]: This looks good to me. Thank you!

**Review by tjons — APPROVED** [review]

**Review by jmhbh — APPROVED** [review]

**Pull request merged**